api/src/main/java/org/pmiops/workbench/absorb/AbsorbServiceImpl.java

```diff
@@ -0,0 +1,99 @@
+package org.pmiops.workbench.absorb;
+
+import static java.time.format.DateTimeFormatter.ISO_DATE_TIME;
+import static java.util.stream.Collectors.toList;
+
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.util.List;
+import org.pmiops.workbench.absorb.api.AuthenticateApi;
+import org.pmiops.workbench.absorb.api.EnrollmentsApi;
+import org.pmiops.workbench.absorb.api.UsersApi;
+import org.pmiops.workbench.absorb.model.AuthenticationRequest;
+import org.pmiops.workbench.absorb.model.UserCourseEnrollmentResource;
+import org.pmiops.workbench.google.CloudStorageClient;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class AbsorbServiceImpl implements AbsorbService {
+  private final CloudStorageClient cloudStorageClient;
+
+  private String apiKey = null;
+  private String accessToken = null;
+  private String userId = null;
+
+  @Autowired
+  public AbsorbServiceImpl(CloudStorageClient cloudStorageClient) {
+    this.cloudStorageClient = cloudStorageClient;
+  }
+
+  @Override
+  public Boolean userHasLoggedIntoAbsorb(String email) throws ApiException {
```

Comment:
is logged in absorb different from has an account in absorb? By that i mean will this function return false if the user has an account with absorb but is not currently logged into Absorb?

Reply:
> will this function return false if the user has an account with absorb but is not currently logged into Absorb

Close: This function returns false if the user has an account with absorb but **has never** logged into Absorb. I had originally named this "userHasAccountInAbsorb," actually.

However, in our case, a users' account is created the first time they log in, and only then. We may be able to create accounts prior to a user logging in by using the API. Until we do that, the concepts of "logged in before" and "has account" should be equivalent. We should rename the function if that stops being true.

I find this method name clearer to use in other services since it describes the users' actions so far (domain logic) rather than relying on Absorb-specific state (implementation details).

Reply:
Should we add this detail to comment, the function name makes so much sense after reading this

Reply:
Good idea, I added a comment